### ISSUE TITLE
nix-command disabled error within repl

### ISSUE THREAD

**Issue by andir** [NixOS/nix#3516]
I did build Nix from the current master branch and tried to build something. After enabling the `nix-command` feature to access the repl I am being told the `nix-command` feature is disabled. This is confusing as I just enabled it to access the repl:

```shell
$ ./inst/bin/nix --experimental-features "nix-command" repl '<nixpkgs>'
Welcome to Nix version 2.4. Type :? for help.

Loading '<nixpkgs>'...
Added 11745 variables.

nix-repl> :b hello
error: experimental Nix feature 'nix-command' is disabled
```

I saw https://github.com/NixOS/nix/issues/3164 which is somewhat related but I think the error message should be more precise about what is required.

**Comment by regnat** [comment]
Fixed by #4960

**Comment by andir** [comment]
@regnat I don't think this is actually fixed:
```
$ nix repl '<nixpkgs>'
Welcome to Nix version 2.5pre20211007_844dd90. Type :? for help.
Loading '<nixpkgs>'...
Added 15340 variables.
nix-repl> p = callPackage ./. {}
nix-repl> :b p
error: experimental Nix feature 'nix-command' is disabled; use '--extra-experimental-features nix-command' to override
$ nix --version                                                                                     
nix (Nix) 2.5pre20211007_844dd90
$ grep experimental /etc/nix/nix.conf
```

None of the experimental features have been enabled on this machine yet the `repl` is available and building derivations fails. Perhaps the repl should also require that flag as then we never end up in that situation? GIven the that `nix` CLI is subject to changes (until it stabelizes) that might be the correct fix?

**Comment by regnat** [comment]
@andir it should have been fixed by 9ebe02a81e (part of Nix 2.4):

```console
$ NIX_CONFIG='experimental-features =' nix repl '<nixpkgs>'
Welcome to Nix 2.5pre20211130_b96164f. Type :? for help.

Loading '<nixpkgs>'...
warning: Nix search path entry '/home/regnat/Config/overlays' does not exist, ignoring
Added 15309 variables.

nix-repl> :b hello

This derivation produced the following outputs:
  out -> /nix/store/ssccr9khjprx3722cqcjikngzffjl4xf-hello-2.10
```
